TheMET/TheMET/ViewControllers/CatalogViewController.swift

```diff
@@ -21,20 +39,132 @@ class CatalogViewController: UIViewController {
-        newView.translatesAutoresizingMaskIntoConstraints = false
-        self.view.addSubview(newView)
-        NSLayoutConstraint.activate([
-            newView.leadingAnchor.constraint(equalTo: self.view.leadingAnchor, constant: 10),
-            newView.trailingAnchor.constraint(equalTo: self.view.trailingAnchor, constant: -10),
-            newView.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 150),
-//            newView.bottomAnchor.constraint(equalTo: self.view.bottomAnchor, constant: 0)
-        ])
+        
+        loadingCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        failedCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        loadedCatalogView.translatesAutoresizingMaskIntoConstraints = false
+        self.updateContent()
+        self.add(catalogSubview: failedCatalogView, stretchView: true)
+        self.add(catalogSubview: loadedCatalogView, stretchView: false)
+        self.add(catalogSubview: loadingCatalogView, stretchView: true)
+        self.loadCatalogCellDataList { catalogCellDataList in
+            guard let catalogCellDataList = catalogCellDataList else {
+                self.contentStatus = .failed
+                return
+            }
+            self.contentStatus = .loaded(catalogCellDataList)
+        }
+        self.updateContent()
+    }
+    
+    private func updateContent() {
+        switch contentStatus {
+        case .failed:
+            self.loadedCatalogView.content = []
+            self.loadingCatalogView.isHidden = true
+            self.failedCatalogView.isHidden = false
+            self.loadedCatalogView.isHidden = true
+        case .loaded(let catalogCellDataList):
+            self.loadedCatalogView.content = catalogCellDataList
+            self.loadingCatalogView.isHidden = true
+            self.failedCatalogView.isHidden = true
+            self.loadedCatalogView.isHidden = false
+        case .loading:
+            self.loadedCatalogView.content = []
+            self.loadingCatalogView.isHidden = false
+            self.failedCatalogView.isHidden = true
+            self.loadedCatalogView.isHidden = true
+        }
+    }
+   
+    private func add(catalogSubview: UIView, stretchView: Bool) {
+        self.view.addSubview(catalogSubview)
+        if stretchView {
+            NSLayoutConstraint.activate([
+                catalogSubview.leadingAnchor.constraint(equalTo: self.view.leadingAnchor, constant: 0),
+                catalogSubview.trailingAnchor.constraint(equalTo: self.view.trailingAnchor, constant: 0),
+                catalogSubview.topAnchor.constraint(equalTo: self.view.topAnchor, constant: 0),
+                catalogSubview.bottomAnchor.constraint(equalTo: self.view.bottomAnchor, constant: 0)
+            ])
+        } else {
+            NSLayoutConstraint.activate([
```

Comment:
Huge code duplication:
- NSLayoutConstraint.activate() use duplication
- 3 constraints are equal

Pls polish this method

Reply:
fixed